src/app/dataset-view/dataset.service.ts

```diff
@@ -292,8 +240,7 @@ export class AppDatasetService {
                     const datasetInfo =
                         AppDatasetService.getDatasetInfo(datasets);
                     this.searchDatasetInfoChanges(datasetInfo);
-                    this.searchData = datasets.data.tail.content;
-                    this.searchDataChanges(datasets.data.tail.content);
+                    this.datasetOverviewService.changeDatasetOverview(datasets.data.tail.content);
```

Comment:
The data type of the `datasets.data.tail.content` is really weird - it's an array of records like this:

```
export interface SearchHistoryInterface {
    province: string;
    reported_date: string;
    system_time: string;
    total_daily: number;
}
```
It looks this was created once for COVID-related datasets.
This looks like very wrong modeling, we should have an array of abstract cells here, because this heavily depends on the structure of a particular dataset.

Reply:
get rid of the SearchHistoryInterface in the next task